### PR TITLE
Adjust shared_preload_libraries before running pg_upgrade

### PR DESCRIPTION
**Pull request by CyberDem0n**
The old version might not have all libraries available

### CODE REVIEW

File: postgres-appliance/bootstrap/pg_upgrade.py [hunk 1]
```diff
@@ -12,6 +12,47 @@
...
+        for value in shared_preload_libraries.split(','):
+            value = value.strip()
+            if value not in extensions or version >= extensions[value][0] and version <= extensions[value][1]:
+                filtered.append(value)
+        self.config['parameters']['shared_preload_libraries'] = ','.join(filtered)
+
+    def start_old_cluster(self, config, version):
+        self.set_bin_dir(version)
+
+        version = float(version)
```
Comment (avaczi) [inline postgres-appliance/bootstrap/pg_upgrade.py]: If this is needed here, why it is not in the previous function?

File: postgres-appliance/scripts/configure_spilo.py [hunk 2]
```diff
@@ -30,6 +30,16 @@
 MEMORY_LIMIT_IN_BYTES_PATH = '/sys/fs/cgroup/memory/memory.limit_in_bytes'
 
 
+# (min_version, max_version, shared_preload_libraries, extwlist.extensions)
+extensions = {
```
Comment (avaczi) [inline postgres-appliance/scripts/configure_spilo.py]: Wouldn't it make sense to reuse this in `pg_upgrade.py`?

**Comment by avaczi** [comment]
:+1:

**Comment by avaczi** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**